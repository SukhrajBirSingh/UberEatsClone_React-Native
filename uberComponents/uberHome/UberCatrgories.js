import { View, Text, Image, TouchableOpacity } from "react-native";
import React from "react";
import MaterialCommunityIcons from "react-native-vector-icons/MaterialCommunityIcons";

const data = [
  {
    id: 1,
    image: require("../../assets/uberAssets/images/UberX.png"),
    title: "Get a ride",
  },
  {
    id: 2,
    image: require("../../assets/uberAssets/images/uberEats.png"),
    title: "Order food",
  },
];

export default function UberCatrgories({ navigation }) {
  return (
    <View
      style={{
        flexDirection: "row",
        justifyContent: "space-evenly",
        marginHorizontal: 20,
      }}
    >
      {data.map((item, index) => (
        <TouchableOpacity
          onPress={() => {
            if (item.id === 1) {
              navigation.navigate("UberDestination");
            } else {
              navigation.navigate("UberEatsHome");
            }
          }}
          key={index}
        >
          <View
            style={{
              margin: 20,
              backgroundColor: "#eee",
              justifyContent: "center",
              padding: 10,
            }}
          >
            <UberImage image={item.image} />
            <UberInfo title={item.title} />
          </View>
        </TouchableOpacity>
      ))}
    </View>
  );
}

const UberImage = (props) => (
  <View
    style={{ alignItems: "center", justifyContent: "center", marginBottom: 25 }}
  >
    <Image
      source={props.image}
      style={{ height: 120, width: 150 }}
      resizeMode="contain"
    />
  </View>
);

const UberInfo = (props) => (
  <>
    <Text style={{ fontSize: 20, fontWeight: "600" }}>{props.title}</Text>
    <MaterialCommunityIcons
      name="arrow-right-circle"
      size={40}
      style={{ paddingTop: 20 }}
    />
  </>
);
