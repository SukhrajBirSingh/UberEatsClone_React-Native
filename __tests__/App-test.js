import React from "react";
import renderer from "react-test-renderer";
import UberCatrgories from "../uberComponents/uberHome/UberCatrgories";

jest.mock(
  "react-native-vector-icons/MaterialCommunityIcons",
  () => "MockedMaterialCommunityIcons"
);

test("<UberCategories/> renders correctly", () => {
  const tree = renderer.create(<UberCatrgories />).toJSON();
  expect(tree).toMatchSnapshot();
});
