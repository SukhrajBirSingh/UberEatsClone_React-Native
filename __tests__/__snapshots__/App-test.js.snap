// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UberCategories/> renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "marginHorizontal": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#eee",
          "justifyContent": "center",
          "margin": 20,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 25,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../assets/uberAssets/images/UberX.png",
            }
          }
          style={
            Object {
              "height": 120,
              "width": 150,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "600",
          }
        }
      >
        Get a ride
      </Text>
      <MockedMaterialCommunityIcons
        name="arrow-right-circle"
        size={40}
        style={
          Object {
            "paddingTop": 20,
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#eee",
          "justifyContent": "center",
          "margin": 20,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 25,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../assets/uberAssets/images/uberEats.png",
            }
          }
          style={
            Object {
              "height": 120,
              "width": 150,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "600",
          }
        }
      >
        Order food
      </Text>
      <MockedMaterialCommunityIcons
        name="arrow-right-circle"
        size={40}
        style={
          Object {
            "paddingTop": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;
